### PR TITLE
Examine text PDA message shortcut for AIs

### PR DESCRIPTION
**Pull request by Nexusuxen**
[QoL]

## About the PR 

Another AI PR! This one will give AIs a shortcut for messaging PDAs - if a PDA is visible (in one's hands, belt, or ID slots), and its messenger is enabled, the AI will be able to click a link just below the mob examine text to send a direct message! If a PDA does not have an owner swiped into it, it will not render; this is intentional. This is also intended to be exclusive to AIs that are in mainframe or camera view, not cyborgs/piloted shells.

As a note, you cannot directly text a Cyborg in this manner, though with the presence of robotic talk and the `silicon` PDA channel, this isn't a big deal. I might add it sometime though!

Demonstration video, poor quality courtesy of built-in microsoft recording software (ignore the runtimes; they're unrelated to the changes made and are to do with Particles.dm)
https://streamable.com/cqyr67

## Why's this needed? 

A few people have lamented over how long it takes to run the Scan command on one's PDA on highpop. Given how the AI is a fancy computer and has to rely on remote communication instead of speaking in-person, a way to quickly text someone's PDA improves QoL a little. This does not change anything that the AI is capable of, and only streamlines the process of sending texts to people. Additionally, this does NOT interfere with regular PDA texting functions, and only serves as an addition to them.

## Changelog

```changelog
(u)Nexusuxen
(*)AIs may now text people directly from examine text if their PDA (or PDAs) is visible and their messenger active, except if the AI is piloting a shell.
```

### CODE REVIEW

File: code/mob/living/silicon/ai.dm [hunk 1]
```diff
@@ -659,6 +659,12 @@ var/list/ai_emotions = list("Happy" = "ai_happy",\
 		//Might as well log what they said too!
 		logTheThing("diary", src, null, ": [t]", "say")
 
+	if (href_list["net_id"]) // this is triggered by examine() in atom.dm, though you should be able to use this from elsewhere!
+		var/id = href_list["net_id"]
+		var/owner = href_list["owner"]
+		var/message = input(usr, "Please enter message", owner, null)
```
Comment (ZeWaka) [inline code/mob/living/silicon/ai.dm:665]: consider tgui input

Comment (Nexusuxen) [reply code/mob/living/silicon/ai.dm:665]: Currently PDAs use href links and this form of input for sending texts afaik; are you sure? (Would have to take a while to learn TGUI but I can definitely try!)

Comment (ZeWaka) [reply code/mob/living/silicon/ai.dm:665]: No, consider `tgui_input()` for this line.

Comment (Nexusuxen) [reply code/mob/living/silicon/ai.dm:665]: Ah okay!

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**